contracts/metagov/BalancerGaugeStaker.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "../pcv/PCVDeposit.sol";
+import "./utils/LiquidityGaugeManager.sol";
+import "../external/balancer/IBalancerMinter.sol";
+
+/// @title Deposit that can stake in Balancer gauges
+/// @author Fei Protocol
+contract BalancerGaugeStaker is PCVDeposit, LiquidityGaugeManager {
+    address public constant BALANCER_GAUGE_CONTROLLER = 0xC128468b7Ce63eA702C1f104D55A2566b13D3ABD;
+    address public constant BALANCER_MINTER = 0x239e55F427D44C3cc793f49bFB507ebe76638a2b;
+    address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+
+    /// @notice Balancer gauge staker
+    /// @param _core Fei Core for reference
+    constructor(address _core) CoreRef(_core) LiquidityGaugeManager(BALANCER_GAUGE_CONTROLLER) {}
+
+    /// @notice returns total balance of PCV in the Deposit
+    function balance() public view override returns (uint256) {
+        return IERC20(BAL).balanceOf(address(this));
+    }
+
+    /// @notice gets the token address in which this deposit returns its balance
+    function balanceReportedIn() public view virtual override returns (address) {
+        return BAL;
+    }
+
+    /// @notice gets the resistant token balance and protocol owned fei of this deposit
+    function resistantBalanceAndFei() public view virtual override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    /// @notice noop
+    function deposit() external override {}
+
+    /// @notice withdraw BAL held to another address
+    /// the BAL rewards accrue on this PCVDeposit when Gauge rewards are claimed.
+    function withdraw(address to, uint256 amount) external override onlyPCVController whenNotPaused {
+        IERC20(BAL).transfer(to, amount);
```

Comment:
Worthwhile emitting a `Withdraw()` event? I know `transfer` will emit one, but wonder if a client would find it easily to track a specific `Withdraw` event

Reply:
good catch

Reply:
maybe withdraw() can be a noop, too. BAL can be moved with withdrawERC20

Reply:
I'll emit `Withdraw()` when BAL goes out of the deposit (when calling `withdraw()`) and emit `Deposit()` when BAL is claimed on the minter, so that SUM(Deposit)-SUM(Withdrawal) = current balance.